kartothek/serialization/_generic.py

```diff
@@ -167,6 +172,23 @@ def check_predicates(predicates):
                     )
 
 
+def remove_columns_from_predicates(
```

Comment:
the function name, the docs and the parameters are somewhat misleading here. It does not remove the columns that are in `columns` but keeps the columns that are exactly in there. Or the implementation is wrong (`if col in columns`).

Reply:
I renamed the function name and added docs with an example